### PR TITLE
Rover: Motor Bitmask for setting ESC type

### PR DESCRIPTION
**Pull request by magicrub**
fixes https://github.com/ArduPilot/ardupilot/issues/8632

### CODE REVIEW

**Comment by magicrub** [comment]
I'm thinking find_motor_mask() should move into SRV_Channel::find_motor_mask()

**Comment by magicrub** [comment]
I pushed all the work up into libraries

File: libraries/SRV_Channel/SRV_Channel.h [hunk 1]
```diff
@@ -171,6 +171,9 @@ class SRV_Channel {
     // return true if function is for a multicopter motor
     static bool is_motor(SRV_Channel::Aux_servo_function_t function);
 
+    // return true if function is any throttle type
+    static bool is_throttle(SRV_Channel::Aux_servo_function_t function);
```
Comment (WickedShell) [inline libraries/SRV_Channel/SRV_Channel.h]: const

File: libraries/SRV_Channel/SRV_Channels.cpp [hunk 2]
```diff
@@ -248,3 +248,22 @@ void SRV_Channels::push()
     }
 #endif // HAL_WITH_UAVCAN
 }
+
+// return a bitmask of each channel that is a motor or a throttle
+uint16_t SRV_Channels::get_motor_mask()
+{
+    uint16_t motor_bitmask = 0;
```
Comment (WickedShell) [inline libraries/SRV_Channel/SRV_Channels.cpp]: Add a static assert that `NUM_SERVO_CHANNELS` <= 16, otherwise this could be a problem in the future.

**Comment by rmackay9** [comment]
Txs for this.  One thing though is that I don't think we should be pushing the knowledge of which outputs are motors down into the SRV_Channels library.  It's really the AP_UGVMotors library that knows this and it's not shared between vehicles.

**Comment by magicrub** [comment]
@rmackay9 I've wished I've had that info in Plane so I thought it might be nice to make it a library. All the knowledge is contained in a helper. There's no memory in the SVR_Channels library of the result. Only Rover uses it now, but seems helpful to other classes.

File: libraries/SRV_Channel/SRV_Channels.cpp (hunk 2 above)
Comment (magicrub) [reply libraries/SRV_Channel/SRV_Channels.cpp]: done

File: libraries/SRV_Channel/SRV_Channel.h (hunk 1 above)
Comment (magicrub) [reply libraries/SRV_Channel/SRV_Channel.h]: function can't be const, but I at least added a const to the argument for you :)

**Comment by DavidIngraham** [comment]
Would it be possible to expand this to enable disabling a PWM output entirely? This would eliminate the current restriction of which pins can be sued for GPIOs. This would replace BRD_PWM_USE

**Comment by rmackay9** [comment]
@magicrub, .. I don't agree about putting it in SRV_Channel.  The motors libraries already need to know about what functions are used because that's how it decides what motor outputs functions should be used.  I think all this does is add another library that needs to be changed when a new motors library is added.

The function misses the motors used by omni rovers..

**Comment by rmackay9** [comment]
The other PR has resolved this issue so closing, thanks!

**Pull request closed without merging**